sycl/unittests/scheduler/InOrderQueueHostTaskDeps.cpp

```diff
@@ -52,3 +53,43 @@ TEST_F(SchedulerTest, InOrderQueueHostTaskDeps) {
 
   EXPECT_TRUE(GEventsWaitCounter == 1);
 }
+
+constexpr auto DisablePostEnqueueCleanupName =
+    "SYCL_DISABLE_POST_ENQUEUE_CLEANUP";
+
+TEST_F(SchedulerTest, InOrderQueueSubmissionOrder) {
+  unittest::ScopedEnvVar DisabledCleanup{
+      DisablePostEnqueueCleanupName, "1",
+      detail::SYCLConfig<detail::SYCL_DISABLE_POST_ENQUEUE_CLEANUP>::reset};
+
+  sycl::unittest::PiMock Mock;
+  sycl::platform Plt = Mock.getPlatform();
+
+  context Ctx{Plt};
+  queue InOrderQueue{Ctx, default_selector_v, property::queue::in_order()};
+
+  auto HostTaskEvent =
+      InOrderQueue.submit([&](sycl::handler &CGH) { CGH.host_task([] {}); });
+  auto Kernel1Event = InOrderQueue.submit(
+      [&](sycl::handler &CGH) { CGH.single_task<TestKernel<>>([] {}); });
```

Comment:
Nit; you could just use the queue shortcut to make this shorter.
```suggestion
  auto Kernel1Event = InOrderQueue.single_task<TestKernel<>>([] {});
```
Same goes for the other submits.